jormungandr/src/network/p2p/comm.rs

```diff
@@ -441,11 +441,10 @@ impl Peers {
                         Err(e) => {
                             debug!(
                                 self.logger,
-                                "propagation to peer {} failed: {}",
-                                id,
-                                e.kind()
+                                "propagation to peer failed, unsubscribing peer";
+                                "node_id" => %id,
+                                "reason" => %e.kind()
```

Comment:
on the same spirit it would be good to leverage the `std::error::Error` trait and have a way to display the `source` of the error too if any (recursively). This would be useful for the whole node actually.

Reply:
slog does not have error chain utilities out of the box, and it would probably be messy to marshal the chain into a single message (splitting across messages is even worse). Dealing with dynamically created metadata in slog is cumbersome, it's really optimized for instantiating everything in the log site with macros.

Reply:
This makes duplicate keys in single message almost inevitable. Slog is really dumb when it comes to that, e.g. JSON output doesn't deduplicate and generates malformed output, human readable, but impossible to machine-process.

Reply:
Our long term solution should be `tracing` (#997).